src/test/blockfilter_tests.cpp

```diff
@@ -32,4 +36,37 @@ BOOST_AUTO_TEST_CASE(gcsfilter_test)
     }
 }
 
+BOOST_AUTO_TEST_CASE(blockfilter_basic_test)
+{
+    CBlock block = getBlock13b8a();
+    BlockFilter block_filter(BlockFilterType::BASIC, block);
+    const GCSFilter& filter = block_filter.GetFilter();
+
+    // TXID of first non-coinbase tx in block.
+    const uint256& txid = uint256S("f9fc751cb7dc372406a9f8d738d5e6f8f63bab71986a39cf36ee70ee17036d07");
+    GCSFilter::Element txid_element(txid.begin(), txid.end());
+    BOOST_CHECK(filter.Match(txid_element));
+
+    // Outpoint spent by first non-coinbase tx in block.
+    COutPoint prevout(uint256S("36e8f98c5f5733f88ca00dfa05afd7af5dc34dda802790daba6aa1afcb8c6096"), 0);
+    GCSFilter::Element prevout_element;
+    CVectorWriter(SER_NETWORK, 0, prevout_element, 0, prevout);
+    BOOST_CHECK(filter.Match(prevout_element));
```

Comment:
Maybe also check that `prevout(36e, 1)` is not a match while at it.

Reply:
Good idea.